Sources/CouchDB/CouchDBUtils.swift

```diff
@@ -76,7 +76,35 @@ class CouchDBUtils {
         }
         return nil
     }
-    
+
+    // A set of characters that are valid in requests.
+    // Fix to Kitura-CouchDB issue #104(https://github.com/IBM-Swift/Kitura-CouchDB/issues/104).
+    // CouchDB Agent creates the documents with unescaped `id` or `attachmentname` etc with `+` but expects
+    // the the character to be escaped while retrieving as documented in CouchDB issues:
+    // https://github.com/apache/couchdb/issues/2144 and https://github.com/apache/couchdb/issues/1830
+    // Workaround is to percent encode `+`. Following `customAllowedCharacterSet` variable escapes `+` in url.
+
+    private static let customAllowedCharacterSet =  NSCharacterSet(charactersIn:"\"#%/<>?@\\^`{|}+ ").inverted
+
+    /**
+     Transform the URL into escaped characters.
+
+     - note: URLs can only be sent over the Internet using the ASCII character set, so character escaping will
+     transform unsafe ASCII characters with a '%' followed by two hexadecimal digits.
```

Comment:
This whole comment could simply be:
> Applies percent-encoding to a string, suitable for use as a CouchDB document path. In addition to characters normally encoded in a URL path, instances of the `+` character are also percent-encoded.

I don't think we need to explain why percent-encoding must be performed or its implementation - that is part of the HTTP specification.